### ISSUE TITLE
expose `Shader`s' `set_default_texture_parameter` as a parameter in the inspector

### ISSUE THREAD

**Issue by mieldepoche** [godotengine/godot-proposals#7701]
### Describe the project you are working on

stuff

### Describe the problem or limitation you are having in your project

`Shader` has this method: `set_default_texture_parameter` that allows providing a default value for `Sampler2D` uniforms. It can only be done via code afaik, so you can't provide default values for your editor-made shaders.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

When the shader's info are shown in the Inspector, expose a list of the sampler2d uniforms, each expecting a `Texture2D` resource that can be set and acts as the default value for the uniform.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

```glsl
group_uniforms textures;
uniform sampler2D albedo : source_color;
uniform sampler2D normal;
uniform sampler2D orm;
uniform sampler2D emission;
```
would show, **in a foldable section titled "default uniform values"** or something like that:

![image](https://github.com/godotengine/godot-proposals/assets/71937306/57772262-3255-467b-a4db-ccc17556bf26)

### If this enhancement will not be used often, can it be worked around with a few lines of script?

it would be used relatively often.

### Is there a reason why this should be core and not an add-on in the asset library?

it's about exposing an existing feature to make it useful

**Comment by Calinou** [comment]
I think this should be provided with `: hint` syntax, rather than as something that can't be saved as part of the `.gdshader`. Don't we have `: hint_black`, `: hint_white` and `: hint_transparent` already?

**Comment by mieldepoche** [comment]
yep I wanted to avoid string refs but I guess there's no other way

**Comment by MatrixDev** [comment]
It would be also great if we can set default value for noise from the shader itself. most of the time there is no reason to provide different noises to the the same shader, so it would be a great usability improvement.

Something like this would be really awesome:
```
uniform sampler2D noise : source_noise(128, 128), hint_noise_seamless;
```

This is currently possible to in the visual shader, but not in the regular one.
